compiler/compiler.py

```diff
@@ -262,41 +300,34 @@ def _store_compilation_result(status: CompilationStatus,
         except Exception as e:  # TODO: look at exceptions in object store.
             raise RuntimeError('Failed to store result') from e
     store.set_status(status)
+    logger.debug('_store_compilation_result: ok')
 
 
 # TODO: rename []_dvips_flag parameters when we figure out what they mean.
 # TODO: can we get rid of any of these?
-def compile_source(source: SourcePackage,
-                   output_format: str = 'pdf', add_stamp: bool = True,
-                   timeout: int = 600, add_psmapfile: bool = False,
-                   P_dvips_flag: bool = False, dvips_layout: str = 'letter',
-                   D_dvips_flag: bool = False,
-                   id_for_decryption: Optional[str] = None,
-                   tex_tree_timestamp = None,
-                   verbose: bool = True) \
-        -> Tuple[Optional[str], Optional[str], Optional[str]]:
+def _run(source: SourcePackage, output_format: Format = Format.PDF,
+         add_stamp: bool = True, timeout: int = 600,
+         add_psmapfile: bool = False, P_dvips_flag: bool = False,
+         dvips_layout: str = 'letter', D_dvips_flag: bool = False,
+         id_for_decryption: Optional[str] = None,
+         tex_tree_timestamp: Optional[str] = None,
+         verbose: bool = True) -> Tuple[Optional[str], Optional[str]]:
     """Compile a TeX source package."""
     # We need the path to the directory container the source package on the
     # host machine, so that we can correctly mount the volume in the
     # converter container. We are assuming that the image is not running in
     # the same container as this worker application.
-    source_dir, fname = os.path.split(source.stream)
+    source_dir, fname = os.path.split(source.path)
     image = current_app.config['COMPILER_DOCKER_IMAGE']
     host_source_root = current_app.config['HOST_SOURCE_ROOT']
     container_source_root = current_app.config['CONTAINER_SOURCE_ROOT']
     leaf_path = source_dir.split(container_source_root, 1)[1].strip('/')
     host_source_dir = os.path.join(host_source_root, leaf_path)
-    logger.debug('source_dir: %s', source_dir)
-    logger.debug('host_source_root: %s', host_source_root)
-    logger.debug('container_source_root: %s', container_source_root)
-    logger.debug('leaf_path: %s', leaf_path)
-    logger.debug('host_source_dir: %s', host_source_dir)
-    logger.debug('got image %s', image)
 
     args = [
         '-S /autotex',
-        f'-p {source.source_id}',
-        f'-f {output_format}',  # This doesn't do what we think it does.
+        f'-p 1901.00123',   # {source.source_id}
```

Comment:
Note that this needs to be changed later - I'd flag with a `# TODO:` so we don't miss it!